rotkehlchen/exchanges/bitstamp.py

```diff
@@ -295,18 +315,31 @@ def query_online_deposits_withdrawals(
         # even more fun. If somehow the endpoint is not called in order then we
         # may end up with some crypto asset movements here that would need to
         # check for corresponding asset movement in the DB.
-        serialized_location = Location.BITSTAMP.serialize_for_db()
         indices_to_delete = []
         with self.db.user_write() as write_cursor:
             for idx, crypto_movement in enumerate(crypto_asset_movements):
-                write_cursor.execute(
-                    'SELECT id from asset_movements WHERE location=? AND category=? AND timestamp=? AND asset=?',  # noqa: E501
-                    (serialized_location, crypto_movement.category.serialize_for_db(), crypto_movement.timestamp, crypto_movement.asset.identifier),  # noqa: E501
+                query_result = history_db.get_history_events(
```

Comment:
same here for the query you are doing as I said above. Also isn't this logic repeating from above?

Reply:
Also changed this to do a direct DB query, although it can't just retrieve the identifier as it did before since we need to actually deserialize the event for use with the `edit_event_extra_data` function.

> isn't this logic repeating from above?

not exactly - here its trying to combine the crypto_asset_movements with ones already saved to the database, while above its combining with the ones it just queried.